components/core/src/clp_s/clp-s.cpp

```diff
@@ -269,11 +270,16 @@ int main(int argc, char const* argv[]) {
             return 1;
         }
 
-        clp_s::JsonConstructorOption option;
+        clp_s::JsonConstructorOption option{};
         option.output_dir = command_line_arguments.get_output_dir();
         option.ordered = command_line_arguments.get_ordered_decompression();
         option.archives_dir = archives_dir;
         option.ordered_chunk_size = command_line_arguments.get_ordered_chunk_size();
+        if (false == command_line_arguments.get_mongodb_uri().empty()) {
+            option.metadata_db
+                    = {command_line_arguments.get_mongodb_uri(),
+                       command_line_arguments.get_mongodb_collection()};
```

Comment:
To use this syntax safely, we should define a constructor in `MetadataOption`; otherwise someone could accidentally change the order of the members in `MetadataOption` and it wouldn't be detected until runtime.